cpp/src/strings/replace/replace_re.cu

```diff
@@ -115,7 +115,7 @@ std::unique_ptr<column> replace_re(
   string_view d_repl(replacement.data(), replacement.size());
 
   // compile regex into device object
-  auto d_prog = reprog_device::create(pattern, flags, stream);
+  auto d_prog = reprog_device::create(pattern, flags, capture_groups::NON_CAPTURE, stream);
```

Comment:
Just for my understanding: `replace_re` doesn't allow using captured groups in `replacement`, correct?

Reply:
That's right. 
There is a [`replace_with_backrefs`](https://docs.rapids.ai/api/libcudf/stable/group__strings__replace.html#ga5e36a14e504147cb4cc8c766fa17f97a) API that does use the captured groups.